sources/quest_2/chapter_3/step_14/step_14_target.move

```diff
@@ -0,0 +1,71 @@
+use std::signer;
+use std::string::{String,utf8};
+
+module robinson::my_shore {
+
+    
+    struct GlobalData has key {
+        nb_tree: u8,
+        has_river: bool,
+        shore_location: address,
+        animal: vector<u8>,
+        island_name: String,
+    }
+
+    struct House has store, drop{
+        no_of_members : u8,
+        house_location : address,
+        house_owner : address
+    }
+
+    fun init_house(s: signer, members: u8, location: address) {
+        assert!(signer::address_of(&s) == @0x42, 0);
+        let _house = House{
+            no_of_members : members,
+            house_location : location,
+            house_owner:signer::address_of(&s),
+        };
+       
+    }
+
+    fun add_trees(data: &mut GlobalData) {
+
+        data.nb_tree = data.nb_tree + 1
+    }
+
+    struct Resources has store, drop{
+    food: u64,
+    log: u64
+}
+
+    fun init_resource() {
+        let _resources = Resources {
+        food: 1,
+        log: 2
+    };
+}
+
+    fun resources_avail( r: &Resources ) : (u64, u64) {
+        let (food, log) = (r.food, r.log);
+        (food, log);
+    }
+
+    fun cut_tree(tree: &mut GlobalData, r: &mut Resources){
+
+        tree.nb_tree = tree.nb_tree - 1;
+        r.log = r.log * 5;
+        r.food = r.food / 2;
+    }
+    
+    fun init_GlobalData(){
```

Comment:
There's something missing here in what we want to do, and as it stands it won't compile!